File: TP1/arquivo/arqui.c
#include <stdio.h>
#include <stdlib.h>

void lerE(int total);
void abrirC(int total);

int main()
{
    int numr;
    scanf("%d/n", &numr);
    lerE(numr);
    abrirC(numr);
    return 0;
}

void lerE(int total)
{
    FILE *arquivo = fopen("arq.txt", "w+");
    for (int i = 0; i < total; i++)
    {
        double numlido;
        scanf("%lf", &numlido);
        fprintf(arquivo, "%f\n", numlido);
    }
    fclose(arquivo);
}

void abrirC(int total)
{

    FILE *arquivo = fopen("arq.txt", "r");
    for (int i = total; i >= 0; i--)
    {
    fseek(arquivo, i, SEEK_SET);
        float valor;
        fscanf(arquivo, "%f", &valor);
        if (valor == (long)valor)
        {
            printf("%f\n", (int)valor);
        }
        else
        {
            printf("%f\n", valor);
        }
    }

    fclose(arquivo);
}

File: TP1/arquivo/arqui.java
import java.io.File;
import java.io.IOException;
import java.io.RandomAccessFile;

class arqui {

    public static void main(String[] args) {
        int numR;
        numR = MyIO.readInt();
        
        try{
            lerE(numR);
            abrirC();

        }catch(Exception e){
            MyIO.println("erro");
        }
    }
    
    static void lerE(int total) throws IOException {
        RandomAccessFile arquivo = new RandomAccessFile("arq.txt", "rw");
        for(int i = 0; i < total;i ++){
            double numlido;
            numlido = MyIO.readDouble();
            arquivo.writeDouble(numlido);
        }
        arquivo.close();
    }

    static void abrirC() throws IOException {
        RandomAccessFile arquivo = new RandomAccessFile("arq.txt", "r");
        long n = arquivo.length();
        arquivo.seek(n);
        while (n > 0) {
            n -= 8;
            arquivo.seek(n);
            double valor = arquivo.readDouble();
            if(valor == (long) valor){
                MyIO.println((int) valor);
            }else{
                MyIO.println(valor);
            }
        }

        arquivo.close();
    }
}